src/api/tmc.ts

```diff
@@ -65,4 +67,123 @@ export default class TMC {
+    }
+
+    /**
+     * Requires an organization to be selected
+     * @returns a list of courses belonging to the currently selected organization
+     */
+    public async getCourses(): Promise<Course[] | TMCApiError | ConnectionError> {
+        const orgSlug = this.storage.getOrganizationSlug();
+        if (!orgSlug) {
+            throw new Error("Organization not selected");
+        }
+        return this.tmcApiRequest("core/org/" + orgSlug + "/courses") as
+            Promise<Course[] | TMCApiError | ConnectionError>;
+    }
+
+    /**
+     * @param id course id
+     * @returns a detailed description for the specified course
+     */
+    public async getCourseDetails(id: number): Promise<CourseDetails | TMCApiError | ConnectionError> {
+        return this.tmcApiRequest("core/courses/" + id.toString()) as
+            Promise<CourseDetails | TMCApiError | ConnectionError>;
+    }
+
+    /**
+     * Performs an HTTP request to the hardcoded TMC server
+     * @param endpoint target API endpoint
+     * @param method HTTP method, defaults to GET
+     */
+    private async tmcApiRequest(endpoint: string, method?: "get" | "post"): Promise<TMCApiResponse> {
+        let request = {
+            headers: {},
+            method: method ? method : "get",
+            url: "https://tmc.mooc.fi/api/v8/" + endpoint,
+        };
+
+        if (this.token) {
+            request = this.token.sign(request);
+        }
+
+        try {
+            const response = await fetch.default(request.url, request);
+            if (response.ok) {
+                return response.json();
+            } else {
+                return {statusCode: response.status, statusText: response.statusText};
+            }
+        } catch (error) {
+            return error as ConnectionError;
+        }
+    }
 }
+
+type Course = {
+    id: number;
+    name: string;
+    title: string;
+    description: string;
+    details_url: string;
+    unlock_url: string;
+    reviews_url: string;
+    comet_url: string;
+    spyware_urls: string[];
+};
+
+type CourseDetails = {
+    course: Course & {
+        unlockables: string[],
+        exercises: Exercise[],
+    };
+};
+
+// This is could be missing some fields, doesn't entirely match the description in the API docs
+type Exercise = {
+    id: number;
+    name: string;
+    locked: boolean;
+    deadline_description: string | null;
+    deadline: string | null;
+    soft_deadline: string | null;
+    soft_deadline_description: string | null;
+    checksum: string;
+    return_url: string;
+    zip_url: string;
+    returnable: boolean;
+    requires_review: boolean;
+    attempted: boolean;
+    completed: boolean;
+    reviewed: boolean;
+    all_review_points_given: boolean;
+    memory_limit: number | null;
+    runtime_params: string[];
+    valgrind_strategy: string;
+    code_review_requests_enabled: boolean;
+    run_tests_locally_action_enabled: boolean;
+};
```

Comment:
`type Exercise = {
    id: number;
    name: string;
    locked: boolean;
    deadline_description: string | null;
    deadline: string | null;
    soft_deadline: string | null;
    soft_deadline_description: string | null;
    checksum: string;
    return_url: string;
    zip_url: string;
    returnable: boolean;
    requires_review: boolean;
    attempted: boolean;
    completed: boolean;
    reviewed: boolean;
    all_review_points_given: boolean;
    memory_limit: number | null;
    runtime_params: string[];
    valgrind_strategy: string;
    code_review_requests_enabled: boolean;
    run_tests_locally_action_enabled: boolean;
    exercise_submissions_url: string;
    latest_submission_url: string;
    latest_submission_id: number;
    solution_zip_url: string
};`

Reply:
added missing fields to type Exercise.